src/Device/Driver/AR62xx.cpp

```diff
@@ -0,0 +1,670 @@
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+}
+*/
+
+/*
+ * The driver is derived from the KRT2-driver. This version has implemented two methods yet:
+ * setting the active frequency
+ * setting the passive frequency
+ * It can be added:
+ * setting/reading dual scan on/off
+ * reading frequency change on the radio
+ * setting/reading squelsh settings
+ *
+ */
+
+#include "Device/Driver/AR62xx.hpp"
+#include "Device/Driver.hpp"
+#include "Device/Port/Port.hpp"
+#include "NMEA/Info.hpp"
+#include "RadioFrequency.hpp"
+#include "Thread/Cond.hxx"
+#include "Thread/Mutex.hxx"
+#include "Util/CharUtil.hxx"
+#include "Util/StaticFifoBuffer.hxx"
+#include "LogFile.hpp"
+
+#include <stdint.h>
+#include <stdio.h>
+#include <chrono>
+#include <cstdlib>
+
+//Einige Konstanten und Ausdruecke, das AR62xx hat ein Binaerprotokoll
+constexpr uint8_t HEADER_ID = 0xA5;
+#define PROTID  0x14;
+#define QUERY   BIT(7)
+#define DUAL    BIT(8)
+#define SQUELCH BIT(7)
+#define MAX_CMD_LEN 128
+#define ACTIVE_STATION  1
+#define PASSIVE_STATION 0
+#define RoundFreq(a) ((int)((a)*1000.0+0.5)/1000.0)
+#define Freq2Idx(a) (int)(((a)-118.0) * 3040/(137.00-118.0)+0.5)
+#define BIT(n) (1 << (n))
+#define NAME_SIZE 30
+
+typedef union {
+  uint16_t intVal16;
+  uint8_t  intVal8[2];
+} IntConvertStruct;
+
+typedef struct {
+  double ActiveFrequency;    //active station frequency
+  double PassiveFrequency;   // passive (or standby) station frequency
+  TCHAR PassiveName[NAME_SIZE] ; // passive (or standby) station name
+  TCHAR ActiveName[NAME_SIZE] ;  //active station name
+  int Volume ;               // Radio Volume
+  int Squelch ;              // Radio Squelch
+  int Vox ;                  // Radio Intercom Volume
+  bool Changed;              // Parameter Changed Flag            (TRUE = parameter changed)
+  bool Enabled;              // Radio Installed d Flag            (TRUE = Radio found)
+  bool Dual;                 // Dual Channel mode active flag     (TRUE = on)
+  bool Enabled8_33;          // 8,33kHz Radio enabled             (TRUE = 8,33kHz)
+  bool RX;                   // Radio reception active            (TRUE = reception)
+  bool TX;                   // Radio transmission active         (TRUE = transmission)
+  bool RX_active;            // Radio reception on active station (TRUE = reception)
+  bool RX_standy;            // Radio reception on passive        (standby) station
+  bool lowBAT;               // Battery low flag                  (TRUE = Batt low)
+  bool TXtimeout;            // Timeout while transmission (2Min)
+} Radio_t;
+
+IntConvertStruct CRC;
+IntConvertStruct sFrequency;
+IntConvertStruct sStatus;
```

Comment:
What kind of messy global variables are those? And why are they exported?

Reply:
removed messy global variables